src/main/webapp/profileScript.js

```diff
@@ -446,18 +446,16 @@ function displaySavedInterests(user, viewer) {
 
   if (viewer === PROFILE_VIEWER_PERSONAL || viewer === PROFILE_VIEWER_BUDDY) {
     fetch('/interest').then(response => response.json()).then((interests) => {
-      let interestCount = 0;
-      for (let i = 0; i < interests.length; i ++) {
-        if (interests[i].interestedUsers.includes(user.id)) {
-          interestsList.appendChild(createInterest(interests[i]));
-          interestCount ++;
-        }
-      }
-      if (interestCount == 0) {
+      if (interests.length == 0) {
```

Comment:
I am not sure I am understanding what the interest servlet does correctly. Does it return an object that includes a list of users interested in that location, or does it return the locations the current user is interested in? 
If it is the first case, I don't think this code change will work. Say a user has no saved interests, but other users do. This would not append the no interests message to the profile of the user who has no saved interests, because the length of the array of all saved interests is not zero. 
Plz ignore if I misunderstood tho

Reply:
To my eye it looks like the servlet responds with a list of all interests in the datastore.

https://github.com/googleinterns/step137-2020/blob/ca02167ff31ead55282cff31bd723482e620e17c/src/main/java/com/google/sps/servlets/InterestServlet.java#L57-L77

Reply:
Oh wait yeah, you're definitely right! Never mind, I should have had confidence in the thought processes of Past Vivian.

Reply:
Whoops, that's my bad. I totally missed that there was some filtering going on in the original diff. 😅